centrallix/expression/exp_functions.c

```diff
@@ -4159,6 +4161,88 @@ int exp_fn_cos_compare(pExpression tree, pParamObjects objlist, pExpression i0,
     return 0;
     }
 
+
+/*
+ * exp_fn_argon2id
+ * This method hashes a given password using the Argon2 algorithm (ID variant)
+ *
+ * Parameters:
+ * 	pExpression tree: 
+ * 	pParamObjects: 
+ * 	pExpression passowrd: The password, passed as a pExpression
+ * 	pExpression salt: The salt, passed as a pExpression
+ *
+ * returns:
+ *	 0 if successful
+ *	 -1 if error
+ */
+int exp_fn_argon2id(pExpression tree, pParamObjects objlist, pExpression password, pExpression salt)
+{   
+    // Set the parameters for the Argon function that are not passed in
+    int HASHLEN = 32;
+    unsigned int T_COST = 2;
+    unsigned int M_COST = (1 << 16);
+    unsigned int PARALLELISM = 1;
```

Comment:
Is there some documentation on how these three values were chosen and what their typical impact is on the hash runtime and memory use?

(future direction: allow the 3rd and 4th params, optional, to specify time/mem costs; btw beyond the 3rd param you have to access it via tree->Children.)

Reply:
To record what we talked about in person:
Those values were simply the defaults in the C example implementation. The actual values we use should be tuned to our particular system. 

I've added support for optional parameters with default values.